pkg/apis/hco/v1beta1/hyperconverged_types_test.go

```diff
@@ -301,200 +301,89 @@ var _ = Describe("HyperconvergedTypes", func() {
 	})
 
 	Context("HyperConvergedFeatureGates", func() {
+		disabled := false
+		enabled := true
 
-		Context("Test IsHotplugVolumesEnabled", func() {
-			It("Should return false if HyperConvergedFeatureGates is nil", func() {
+		Context("Test RebuildEnabledGateMap", func() {
+			It("Should return nil if HyperConvergedFeatureGates is nil", func() {
 				var fgs *HyperConvergedFeatureGates = nil
-				Expect(fgs.IsHotplugVolumesEnabled()).To(BeFalse())
+				Expect(fgs.RebuildEnabledGateMap()).To(BeNil())
 			})
 
 			It("Should return false if HotplugVolumes does not exist", func() {
 				fgs := &HyperConvergedFeatureGates{}
-				Expect(fgs.IsHotplugVolumesEnabled()).To(BeFalse())
+				m := fgs.RebuildEnabledGateMap()
```

Comment:
You know these are not equal in two ways. We are over-checking here. I think we should go back to the old way.